src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    public Ball getRandomBall() {
+        ColorSupplier colorSupplier = new ColorSupplier();

```

Comment:
ColorSupplier object creates every time you call getRandomFigure/getDefaultFigure. Think of approach, when you create an object of ColorSupplier only once for this class

Reply:
Greate! You have created colorSupplier a class level field, but you still using new ColorSupplier each time you call this method. Just use your new created field